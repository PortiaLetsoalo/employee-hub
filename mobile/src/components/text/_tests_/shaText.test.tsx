import React from 'react';
import { create, act } from 'react-test-renderer';
import ShaText from '../shaText';

const tree = create(
  <ShaText style={{ backgroundColor: 'red' }}>This is a test</ShaText>
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('shaText isDefined', () => {
  expect(<ShaText />).toBeDefined();
});

test('shaText render', () => {
  expect(tree.root.props.children).toEqual('This is a test');
  expect(tree.root.props.style.backgroundColor).toEqual('red');

  act(() => {
    tree.update(
      <ShaText style={{ backgroundColor: 'green' }}>
        <ShaText>This is a nested ShaText</ShaText>
      </ShaText>
    );
  });
  expect(tree.root.props.children).toEqual(
    <ShaText>This is a nested ShaText</ShaText>
  );
  expect(tree.root.props.style.backgroundColor).toEqual('green');
});
