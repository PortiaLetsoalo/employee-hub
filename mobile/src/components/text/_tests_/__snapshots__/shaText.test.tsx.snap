// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<Text
  style={
    Object {
      "backgroundColor": "red",
      "color": undefined,
      "fontFamily": undefined,
      "fontSize": 14,
    }
  }
>
  This is a test
</Text>
`;
